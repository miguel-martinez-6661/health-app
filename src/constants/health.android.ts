import {HealthState} from '../interface/health.interface';
import {Scopes, BucketUnit} from 'react-native-google-fit';

export const initialState: HealthState = {
  permissionGranted: false,
  stepCount: 0,
};

export const options = {
  scopes: [Scopes.FITNESS_ACTIVITY_READ, Scopes.FITNESS_BODY_READ],
};

export const googleFitOptions = {
  startDate: '2017-01-01T00:00:17.971Z',
  endDate: new Date().toISOString(),
  bucketUnit: BucketUnit.DAY,
  bucketInterval: 1,
};
