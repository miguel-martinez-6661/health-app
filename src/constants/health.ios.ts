import AppleHealthKit, {HealthKitPermissions} from 'react-native-health';

export const options = {
  date: new Date().toISOString(), // optional; default now
  includeManuallyAdded: false, // optional: default true
};

/* Permission options */
export const permissions = {
  permissions: {
    read: [AppleHealthKit.Constants.Permissions.HeartRate],
    write: [AppleHealthKit.Constants.Permissions.Steps],
  },
} as HealthKitPermissions;
