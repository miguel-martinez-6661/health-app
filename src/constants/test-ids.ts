export const TestIDs = {
  HEADER: 'HEADER',
  HEADER_GREETINGS: 'HEADER_GREETINGS',
  HEADER_TITLE: 'HEADER_TITLE',

  HEALTH_STEP_SECTION: 'HEALTH_STEP_SECTION',
  HEALTH_STEP_NAME: 'HEALTH_STEP_NAME',
  HEALTH_STEP_VALUE: 'HEALTH_STEP_VALUE',
  HEALTH_STEP_DESCRIPTION: 'HEALTH_STEP_DESCRIPTION',

  REQUEST_PERMISSION_SECTION: 'REQUEST_PERMISSION_SECTION',
  REQUEST_PERMISSION_BUTTON: 'REQUEST_PERMISSION_BUTTON',
};
