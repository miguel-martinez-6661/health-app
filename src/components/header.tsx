import React from 'react';
import {Text, View} from 'react-native';
import {TestIDs} from '../constants/test-ids';

export const Header = () => {
  return (
    <View testID={TestIDs.HEADER} className="p-2">
      <Text testID={TestIDs.HEADER_GREETINGS} className="text-lg">
        Hello, Miguel
      </Text>
      <Text testID={TestIDs.HEADER_TITLE} className="font-bold text-3xl">
        Activities
      </Text>
    </View>
  );
};
