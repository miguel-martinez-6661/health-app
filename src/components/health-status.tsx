import React from 'react';
import {Text, View} from 'react-native';
import {TestIDs} from '../constants/test-ids';

interface WithChildren {
  children: React.ReactNode;
  description?: string;
}

const HealthItemName = ({children}: WithChildren) => {
  return (
    <Text
      testID={TestIDs.HEALTH_STEP_NAME}
      className="text-lg font-bold text-white">
      {children}
    </Text>
  );
};

const HealthItemValue = ({children, description}: WithChildren) => {
  return (
    <View className="flex-row items-end gap-2">
      <Text
        testID={TestIDs.HEALTH_STEP_VALUE}
        className="text-5xl text-white font-thin">
        {children}
      </Text>
      <Text
        testID={TestIDs.HEALTH_STEP_DESCRIPTION}
        className="text-lg text-white pb-1 font-thin">
        {description}
      </Text>
    </View>
  );
};

export const HealthStepStatus = ({children}: WithChildren) => {
  return (
    <View testID={TestIDs.HEALTH_STEP_SECTION} className="m-2">
      <Text className="my-2 text-lg">Your Information</Text>
      <View className="p-2 flex-row justify-between rounded-lg bg-red-500 h-28 ">
        {children}
      </View>
    </View>
  );
};

HealthStepStatus.Title = HealthItemName;
HealthStepStatus.Value = HealthItemValue;
