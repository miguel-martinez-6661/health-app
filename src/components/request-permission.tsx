import React from 'react';
import {Text, TouchableOpacity, View} from 'react-native';
import {TestIDs} from '../constants/test-ids';

interface RequestPermissionProps {
  onPress: () => void;
}

export const RequestPermission = ({onPress}: RequestPermissionProps) => {
  return (
    <View
      testID={TestIDs.REQUEST_PERMISSION_SECTION}
      className="bg-blue-400 m-2 p-4 rounded-lg">
      <Text className="text-lg">
        To start tracking your Health info, you should grant permissions
      </Text>
      <TouchableOpacity
        testID={TestIDs.REQUEST_PERMISSION_BUTTON}
        className="bg-blue-700 p-2 items-center rounded-lg mt-4"
        onPress={onPress}>
        <Text className="text-white text-lg">Grant Permissions</Text>
      </TouchableOpacity>
    </View>
  );
};
