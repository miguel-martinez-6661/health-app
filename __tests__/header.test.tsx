import React from 'react';
import {render, screen} from '@testing-library/react-native';
import {Header} from '../src/components/header';
import {TestIDs} from '../src/constants/test-ids';

describe('<Header /> Testing', () => {
  it('should match the Header snapshot', () => {
    const {toJSON} = render(<Header />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render the Header component', () => {
    const {getByTestId} = render(<Header />);
    expect(getByTestId(TestIDs.HEADER)).toBeDefined();
  });

  it('should render the Header component with the correct fields', async () => {
    render(<Header />);
    const greetings = await screen.findByTestId(TestIDs.HEADER_GREETINGS);
    const title = await screen.findByTestId(TestIDs.HEADER_TITLE);
    expect(greetings.props.children).toBe('Hello, Miguel');
    expect(title.props.children).toBe('Activities');
  });
});
