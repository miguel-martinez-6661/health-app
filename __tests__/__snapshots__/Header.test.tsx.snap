// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Testing should match the Header snapshot 1`] = `
<View
  style={
    [
      {
        "paddingBottom": 8,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 8,
      },
    ]
  }
  testID="HEADER"
>
  <Text
    style={
      [
        {
          "fontSize": 18,
          "lineHeight": 28,
        },
      ]
    }
    testID="HEADER_GREETINGS"
  >
    Hello, Miguel
  </Text>
  <Text
    style={
      [
        {
          "fontWeight": "700",
        },
        {
          "fontSize": 30,
          "lineHeight": 36,
        },
      ]
    }
    testID="HEADER_TITLE"
  >
    Activities
  </Text>
</View>
`;
